qiita_db/data.py

```diff
@@ -807,6 +807,72 @@ def create(cls, study, preprocessed_params_table, preprocessed_params_id,
         ppd.add_filepaths(filepaths, conn_handler)
         return ppd
 
+    @classmethod
+    def delete(cls, ppd_id):
+        """Removes the preprocessed data with id ppd_id
+
+        Parameters
+        ----------
+        ppd_id : int
+            The preprocessed data id
+
+        Raises
+        ------
+        QiitaDBStatusError
+            If the preprocessed data status is not sandbox or if the
+            preprocessed data EBI and VAMPS submission is not in a valid state
+            ['not submitted', 'failed']
+        QiitaDBError
+            If the preprocessed data has been processed
+        """
+        valid_submission_states = ['not submitted', 'failed']
+        ppd = cls(ppd_id)
+        if ppd.status != 'sandbox':
+            raise QiitaDBStatusError(
+                "Illegal operation on non sandbox preprocessed data")
```

Comment:
Since these messages (832 and 836) are likely to be seen by users on the GUI, do you think you could rephrase them?